sdk/logs/src/test/java/io/opentelemetry/sdk/logs/SdkLogEmitterTest.java

```diff
@@ -104,20 +104,25 @@ void logRecordBuilder_maxAttributes() {
                 () -> LogLimits.builder().setMaxNumberOfAttributes(maxNumberOfAttrs).build())
             .build();
 
-    AttributesBuilder attributesBuilder = Attributes.builder();
+    LogRecordBuilder builder = logEmitterProvider.get("test").logRecordBuilder();
     for (int i = 0; i < 2 * maxNumberOfAttrs; i++) {
-      attributesBuilder.put("key" + i, i);
+      builder.setAttribute(AttributeKey.longKey("key" + i), (long) i);
     }
+    builder.emit();
 
-    logEmitterProvider
-        .get("test")
-        .logRecordBuilder()
-        .setAllAttributes(attributesBuilder.build())
-        .emit();
-
-    // NOTE: cannot guarantee which attributes are retained, only that there are no more that the
-    // max
-    assertThat(seenLog.get().toLogData().getAttributes()).hasSize(maxNumberOfAttrs);
+    assertThat(seenLog.get().toLogData())
+        .hasAttributes(
+            Attributes.builder()
```

Comment:
Nit: you could build an `expectedAttributes` instance along with the log record, and you wouldn't have to hard-code the 8 attributes here